orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationModuleConfig.java

```diff
@@ -211,4 +220,95 @@ public OrchestrationModuleConfig withGrounding(
+    }
+
+    //  Make sure old responseFormat is only overwritten if new templateConfig has one set.
+    var newTemplate = (Template) templateConfig;
+    TemplateResponseFormat responseFormat = null;
+    if (this.templateConfig instanceof Template oldTemplate) {
+      responseFormat = oldTemplate.getResponseFormat();
+    }
+    //  Template.getResponseFormat() might return null, so the following check is necessary.
+    if (newTemplate != null && newTemplate.getResponseFormat() == null) {
+      newTemplate.setResponseFormat(responseFormat);
+    }
+    if (newTemplate == null && responseFormat != null) {
+      newTemplate = Template.create().template(List.of());
+      newTemplate.setResponseFormat(responseFormat);
+    }
+
+    return new OrchestrationModuleConfig(
+        llmConfig, newTemplate, maskingConfig, filteringConfig, groundingConfig);
+  }
+
+  /**
+   * Creates a new configuration with the given response schema.
+   *
+   * @link <a
+   *     href="https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/structured-output">SAP
+   *     AI Core: Orchestration - Structured Output</a>
+   * @param schema The response schema to use.
+   * @return A new configuration with the given response schema.
+   * @since 1.4.0
+   */
+  @Nonnull
+  public OrchestrationModuleConfig withJsonSchemaResponse(
+      @Nonnull final ResponseJsonSchema schema) {
+    val responseFormatJsonSchema =
+        ResponseFormatJsonSchema.create()
+            .type(ResponseFormatJsonSchema.TypeEnum.JSON_SCHEMA)
+            .jsonSchema(
+                ResponseFormatJsonSchemaJsonSchema.create()
+                    .name(schema.getName())
+                    .schema(schema.getSchemaMap())
+                    .strict(schema.getIsStrict())
+                    .description(schema.getDescription()));
+    if (this.templateConfig instanceof Template template) {
+      template.setResponseFormat(responseFormatJsonSchema);
+      return this.withTemplateConfig(template);
+    }
+    val templatingConfig =
+        Template.create().template(List.of()).responseFormat(responseFormatJsonSchema);
+    return this.withTemplateConfig(templatingConfig);
+  }
+
+  /**
+   * Creates a new configuration where the response format is set to JSON object.
+   *
+   * @link <a
+   *     href="https://help.sap.com/docs/sap-ai-core/sap-ai-core-service-guide/structured-output">SAP
+   *     AI Core: Orchestration - Structured Output</a>
+   * @return A new configuration with response format JSON object.
+   * @since 1.4.0
+   */
+  @Nonnull
+  public OrchestrationModuleConfig withJsonResponse() {
+    val responseFormatJsonObject =
+        ResponseFormatJsonObject.create().type(ResponseFormatJsonObject.TypeEnum.JSON_OBJECT);
+    if (this.templateConfig instanceof Template template) {
+      template.setResponseFormat(responseFormatJsonObject);
+      return this.withTemplateConfig(template);
+    }
```

Comment:
(Question/Minor)

This has side-effect doesn't it?

The following code is (non) functional equivalent:
```suggestion
    if (this.templateConfig instanceof Template template) {
      template.setResponseFormat(responseFormatJsonObject);
      return this;
    }
```

Resulting (unexpectedly) in...
```java
val gpt4oCustomInstance = new OrchestrationModuleConfig().withLlmConfig(GPT_4O_MINI);
val configWithJsonResponse = gpt4oCustomInstance.withJsonResponse();

gpt4oCustomInstance.equals(configWithJsonResponse) // true
```